docs/specs/captcha.md

```diff
@@ -1,41 +1,452 @@
+- `captcha.ip_allowlist`: A list of IPv4 CIDR notations or IPv4 addresses. If the incoming request matches any entry in the allowlist, the request bypasses Captcha.
+- `captcha.providers`: A list of Captcha provider configuration. The actual shape depends on the `type` property.
+- `captcha.providers.type`: Required. The type of the Captcha provider. Valid values are `cloudflare` and `recaptchav2`.
+- `captcha.providers.alias`: Required. The unique identifier of the Captcha provider. It is used in other parts of the configuration to refer to this particular Captcha provider. For example, the project can configured a number of Captcha providers, but only uses one of them in a particular Authentication Flow.
+
+
+Type specific fields:
+
+- `captcha.providers.type=cloudflare.site_key`: The site key of Cloudflare Turnstile.
+- `captcha.providers.type=recaptchav2.site_key`: The site key of reCAPTCHA v2.
+
+### authgear.secrets.yaml
 
 ```yaml
 - data:
-    secret: YOUR_TURNSTILE_SECRET_KEY
-  key: captcha.cloudflare
+    items:
+    - type: recaptchav3
+      alias: recaptchav3
+      secret_key: RECAPTCHAV3_SECRET_KEY
+  key: risk_assessment.providers
+
+- data:
+    items:
+    - type: cloudflare
+      alias: cloudflare
+      secret_key: TURNSTILE_SECRET_KEY
+  key: captcha.providers
 ```
 
-- Provider: `cloudflare`
-  - `key`: `captcha.cloudflare`
-  - `data.secret`: The turnstile secret key
+- `key=risk_assessment.providers.items.type`: Required. It is the same as `risk_assessment.providers.type`.
+- `key=risk_assessment.providers.items.alias`: Required. It is the same as `risk_assessment.providers.alias`.
+
+Type specific fields:
+
+- `key=risk_assessment.providers.items.type=recaptchav3.secret_key`: Required. The secret key of reCAPTCHA v3.
+
+---
+
+- `key=captcha.providers.items.type`: Required. It is the same as `captcha.providers.type`.
+- `key=captcha.providers.items.alias`: Required. It is the same as `captcha.providers.alias`.
+
+Type specific fields:
+
+- `key=captcha.providers.items.type=cloudflare.secret_key`: Required. The secret key of Cloudflare Turnstile.
+- `key=captcha.providers.items.type=recaptchav2.secret_key`: Required. The secret key of reCAPTCHA v2.
+
+## Authentication Flow
+
+This section specifies how risk assessment and Captcha works in a Authentication Flow.
+
+### Risk assessment and Captcha in Authentication Flow configuration
+
+Risk assessment and Captcha are supported in the following flow types:
+
+- `signup`
+- `promote`
+- `login`
+- `signup_login`
+- `reauth`
+- `account_recovery`
+
+Risk assessment and Captcha are supported only in the following step types:
+
+- `identify` in `signup`, `promote`, `login`, `signup_login`, and `account_recovery`.
+- `create_authenticator` in `signup` and `promote`.
+- `authenticate` in `login` and `reauth`.
+
+We can see that all supported step types have branches.
+
+To enable risk assessment and Captcha in a branch, add `risk_assessment` and `captcha` to the branch.
```

Comment:
question: For a certain auth flow `flowA`, if one of `flowA`'s branch `branchX` enabled `captcha`, does below endpoints
- create - `POST /api/v1/authentication_flows`
- input - `POST /api/v1/authentication_flows/input`

i. always return `captcha` field; OR
ii. only return `captcha` field when the handler contains `branchX` logic?

```json
"result": {
        "state_token": "authflowstate_blahblah",
        "type": "signup", # "signup" | "login" | ...
        "name": "default",
         // below object
         "captcha": {
            "providers": [
                {
                    "type": "cloudflare",
                    "alias": "cloudflare"
                }
            ]
        },
```

Reply:
I have updated the api reference. We no longer return `captcha` in the root of the response. Instead, the captcha information is returned in the branch.

Reply:
I see, thanks!